### PR TITLE
Fix docstring cross-references

### PR DESCRIPTION
**Pull request by mthuurne**
## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/9873
* [x] The changes pass minimal style checks (see: https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted )
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles )
* [x] ~~I have updated the automated tests.~~ Only docstrings were changed.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.

### CODE REVIEW

File: src/twisted/cred/credentials.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ class ICredentials(Interface):
     I check credentials.
 
     Implementors I{must} specify the sub-interfaces of ICredentials
-    to which it conforms, using L{zope.interface.declarations.implementer}.
```
Comment (rodrigc) [inline src/twisted/cred/credentials.py:36]: Is this change necessary?  `implementer` is actually defined in `zope.interface.declarations`
although `zope.interface` imports it.

**Review by rodrigc — APPROVED** [review]
Minor comment, but otherwise looks ok.

File: src/twisted/cred/credentials.py (hunk 1 above)
Comment (mthuurne) [reply src/twisted/cred/credentials.py:36]: It is defined in `declarations`, but not documented there, so to link to its documentation we need to point to the parent module.

Comment (rodrigc) [reply src/twisted/cred/credentials.py:36]: Ah, OK.

**Comment by rodrigc** [comment]
Thanks for these fixes.  You are now on https://twistedmatrix.com/highscores/ :)
In addition to fixing these docstrings, please consider submitting additional PR's where you add type annotations
to functions.   
That will increase your highscores. :)

**Comment by mthuurne** [comment]
I might do that one day, but for now I'm going to continue working on pydoctor.
